### PR TITLE
Update gallery and finished TDD

### PR DESCRIPTION
**Pull request by gaurnett**
Hi Aditya and Taylor,

I finished the required parts of the TDD and would like for you to take a look. There are a lot of files here but most of them are helper files to complete the job given to us by the intern team. The ones I edited are FindMeetingQuery.java and FindMeetingQueryTest.java.

### CODE REVIEW

File: portfolio/src/main/webapp/js/script.js [hunk 1]
```diff
@@ -13,6 +13,7 @@
 // limitations under the License.
 
 window.onload = function () {
+    filterGallery('all');
```
Comment (taylorwan) [inline portfolio/src/main/webapp/js/script.js:16]: nice!

File: walkthroughs/week-5-tdd/intro/pom.xml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        <artifactId>appengine-maven-plugin</artifactId>
+        <version>2.2.0</version>
+        <configuration>
+          <deploy.projectId>YOUR_PROJECT_ID_HERE</deploy.projectId>
+          <deploy.version>1</deploy.version>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
+</project>
```
Comment (taylorwan) [inline walkthroughs/week-5-tdd/intro/pom.xml]: nit: newline

**Review by taylorwan — COMMENTED** [review]
could you break this out into two PRs, so we can see which code is part of the tutorial, and which parts are your code?

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 3]
```diff
@@ -270,5 +274,186 @@ public void notEnoughRoom() {
```
Comment (mithasgoogle) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:274]: Very detailed and comprehensive test.

Can you try creating a setUp() method where you do a lot of the work that's common to all your tests?  For example, where you're adding to the Event collection?

**Review by mithasgoogle — APPROVED** [review]

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 4]
```diff
@@ -270,5 +274,186 @@ public void notEnoughRoom() {
...
+
+  @Test
+  public void optionalAttendeeWithOneEvent() {
+    // Have each person have different events. We should see two options because each person has
+    // split the restricted times.
+    //
+    // Events  :       |--A--|     |--B--|
+    //                       |--C--|
+    // Day     : |-----------------------------|
+    // Options : |--1--|     |--2--|     |--3--|
```
Comment (taylorwan) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:316]: should the options here just be 1 and 3?

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 5]
```diff
@@ -270,5 +274,186 @@ public void notEnoughRoom() {
...
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void justEnoughRoomForOptionalAttendee() {
+    // Have one person, but make it so that there is just enough room at one point in the day to
+    // have the meeting.
+    //
+    // Events  : |--A--|     |----A----|
+    //                 |-C-|
```
Comment (taylorwan) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:375]: should this say B, or maybe we are missing a person C either in the diagram / the code?

thinking that if the only other person in the meeting is optional and the time slot doesn't work for them - it's only a one person meeting!

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 6]
```diff
@@ -270,5 +274,186 @@ public void notEnoughRoom() {
...
+    Assert.assertEquals(expected, actual);
+  }
+
+    @Test
+    public void optionalAttendeesWithGaps() {
+        // No mandatory attendees, just two optional attendees with several gaps in their 
+        // schedules. Those gaps should be identified and returned.
+        //
+        // Events  :     |-A-|             |-B-|
+        //                        |-C-|
```
Comment (taylorwan) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:403]: should this say A?

**Review by taylorwan — COMMENTED** [review]
Great job with the test cases, and great commenting! Left some comments for a few test cases that look different from the code - should be trivial updates!

**Pull request merged**